woocommerce/woocommerce-ios
# [Mobile Payments] Refresh Payment Gateways when Payment Menu opened

Closes: #7633


### Description

With the Cash on Delivery project, we added a toggle to set the `enabled` property for the `cod` gateway, described as Pay in Person.

This toggle was set based on the stored Payment Gateways, which were only synchronised on a new launch of the app, log in, or when the merchant switched site.

This change adds a `synchronizePaymentGateways` call when the Payments Menu is opened, so that we are showing fresh data for the toggle.

Additionally, the previous `InPersonPaymentsMenuViewModel` has been renamed to show that it's specific to the Pay in Person Toggle row, and a new one added for the menu itself.

### Testing instructions


1. Open the app
2. Change the Enabled state of the `Cash on Delivery` payment method for your store, using wp-admin
3. Navigate to `Menu > Payments`
4. Observe that the toggle is updated to show the current `enabled` state of the payment gateway


### Screenshots



---
